java.lang.NullPointerException starting at 20220129204202 (updated)

## Problem description
I've been running an automatic download/update script of my own for a while now.
Installing 11.0.0-SNAPSHOT.20220129204202, AA fails to stay running.

### Steps to reproduce
systemctl start airsonic.service

## System information
11.0.0-SNAPSHOT.20220129204202
Ubuntu 20.04.3 LTS
openjdk 11.0.13 2021-10-19
No proxy.
English

## Additional notes
From airsonic.log

```
2022-01-30 15:26:02.600 ERROR --- o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'themeSource': Unsatisfied dependency expressed through method 'setSettingsService' parameter 0; nested exception is org.springframework.b
eans.factory.UnsatisfiedDependencyException: Error creating bean with name 'settingsService': Unsatisfied dependency expressed through field 'internetRadioDao'; nested exception is org.springframework.beans.factory.UnsatisfiedDependen
cyException: Error creating bean with name 'internetRadioDao': Unsatisfied dependency expressed through field 'jdbcTemplate'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name '
liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.LiquibaseE
xception: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::populate-podcast-channel-media-file-id::anon:
     Reason: java.lang.NullPointerException
```

Fortunately, this is a VM so I have a snapshot of the last working version (11.0.0-SNAPSHOT.20220124042650).

after the service fails, journalctl -xe yields...
```
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:322) ~[spring-beans-5.3.8.jar!/:5.3.8]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.8.jar!/:5.3.8]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276) ~[spring-beans-5.3.8.jar!/:5.3.8]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1380) ~[spring-beans-5.3.8.jar!/:5.3.8]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1300) ~[spring-beans-5.3.8.jar!/:5.3.8]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:657) ~[s>
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         ... 57 common frames omitted
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]: Caused by: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::populate>
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:      Reason: java.lang.NullPointerException
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:124) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Liquibase.lambda$null$0(Liquibase.java:265) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.lambda$child$0(Scope.java:177) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.child(Scope.java:186) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.child(Scope.java:176) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.child(Scope.java:155) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.child(Scope.java:240) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Liquibase.lambda$update$1(Liquibase.java:264) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.lambda$child$0(Scope.java:177) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.child(Scope.java:186) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.child(Scope.java:176) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.child(Scope.java:155) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Liquibase.runInScope(Liquibase.java:2404) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Liquibase.update(Liquibase.java:211) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Liquibase.update(Liquibase.java:197) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:314) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:269) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1845) ~[spring-beans-5.3.8.jar!/:5.3.>
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1782) ~[spring-beans-5.3.8.jar!/:5.3.8]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         ... 69 common frames omitted
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]: Caused by: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase/11.0/relative-media-file-paths.xml::populate-podcast-channel-media-file-id::anon:
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:      Reason: java.lang.NullPointerException
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.changelog.ChangeSet.execute(ChangeSet.java:695) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:49) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.changelog.ChangeLogIterator$2.lambda$null$0(ChangeLogIterator.java:111) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.lambda$child$0(Scope.java:177) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.child(Scope.java:186) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.child(Scope.java:176) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.child(Scope.java:155) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.changelog.ChangeLogIterator$2.lambda$run$1(ChangeLogIterator.java:110) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.lambda$child$0(Scope.java:177) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.child(Scope.java:186) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.child(Scope.java:176) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.child(Scope.java:155) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.child(Scope.java:240) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.changelog.ChangeLogIterator$2.run(ChangeLogIterator.java:94) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.lambda$child$0(Scope.java:177) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.child(Scope.java:186) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.child(Scope.java:176) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.child(Scope.java:155) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.child(Scope.java:240) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.Scope.child(Scope.java:244) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:66) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         ... 87 common frames omitted
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]: Caused by: java.lang.NullPointerException: null
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at org.airsonic.player.spring.migrations.AddMediaFileIdToPodcastChannels.generateStatements(AddMediaFileIdToPodcastChannels.java:65) ~[classes!/:11.0.0-SNAPSHOT]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.change.custom.CustomChangeWrapper.generateStatements(CustomChangeWrapper.java:162) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.database.AbstractJdbcDatabase.executeStatements(AbstractJdbcDatabase.java:1259) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         at liquibase.changelog.ChangeSet.execute(ChangeSet.java:660) ~[liquibase-core-4.6.2.jar!/:na]
Jan 30 16:45:54 airsonic.siminiuk.com java[1497]:         ... 107 common frames omitted
Jan 30 16:45:54 airsonic.siminiuk.com systemd[1]: airsonic.service: Main process exited, code=exited, status=1/FAILURE
-- Subject: Unit process exited
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- An ExecStart= process belonging to unit airsonic.service has exited.
--
-- The process' exit code is 'exited' and its exit status is 1.
Jan 30 16:45:54 airsonic.siminiuk.com systemd[1]: airsonic.service: Failed with result 'exit-code'.
-- Subject: Unit failed
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- The unit airsonic.service has entered the 'failed' state with result 'exit-code'.
```

--- a/airsonic-main/src/main/java/org/airsonic/player/spring/migrations/UniqueMediaFolders.java
+++ b/airsonic-main/src/main/java/org/airsonic/player/spring/migrations/UniqueMediaFolders.java
@@ -12,6 +12,7 @@ import liquibase.statement.SqlStatement;
 import liquibase.statement.core.DeleteStatement;
 import org.airsonic.player.dao.MusicFolderDao;
 import org.airsonic.player.domain.MusicFolder;
+import org.airsonic.player.domain.MusicFolder.Type;
 import org.airsonic.player.service.MediaFolderService;
 import org.apache.commons.lang3.tuple.Triple;
 import org.slf4j.Logger;
@@ -67,10 +68,16 @@ public class UniqueMediaFolders implements CustomSqlChange {
                     folders.add(MusicFolderDao.MUSICFOLDER_ROW_MAPPER.mapRow(result, i++));
                 }
                 Set<String> paths = new HashSet<>();
+                // skip deleting podcast folder
+                MusicFolder podcastFolder = folders.stream().filter(f -> f.getType() == Type.PODCAST).findFirst().orElse(null);
+                if (podcastFolder == null) {
+                    throw new RuntimeException("Cannot find podcast folder!");
+                }
+                paths.add(podcastFolder.getPath().toString());
                 folders.forEach(f -> {
                     Triple<List<MusicFolder>, List<MusicFolder>, List<MusicFolder>> overlap = MediaFolderService.getMusicFolderPathOverlaps(f, folders);
                     // duplicate
-                    if (!overlap.getLeft().isEmpty() && !paths.add(f.getPath().toString())) {
+                    if (!overlap.getLeft().isEmpty() && !paths.add(f.getPath().toString()) && f.getType() != Type.PODCAST) {
                         deletionSet.add(f.getId());
                         LOG.info("Duplicate media folder found (id: {}, name: {}) and will be deleted", f.getId(), f.getName());
                     }

